From 5e77809a2edf48a5b95b4f019e63d5f83c02cce7 Mon Sep 17 00:00:00 2001
From: Takanari Hayama <taki@igel.co.jp>
Date: Wed, 18 Feb 2015 08:24:28 +0900
Subject: [PATCH 01/12] compositor-drm: map framebuffers with read/write
 permission.

We should map framebuffers with read and write permission. Otherwise, we
won't be able to read back framebuffers when we capture frames.
---
 libweston/compositor-drm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libweston/compositor-drm.c b/libweston/compositor-drm.c
index 38911763..d061c769 100644
--- a/libweston/compositor-drm.c
+++ b/libweston/compositor-drm.c
@@ -1007,7 +1007,7 @@ drm_fb_create_dumb(struct drm_backend *b, int width, int height,
 	if (ret)
 		goto err_add_fb;
 
-	fb->map = mmap(NULL, fb->size, PROT_WRITE,
+	fb->map = mmap(NULL, fb->size, PROT_READ | PROT_WRITE,
 		       MAP_SHARED, b->drm.fd, map_arg.offset);
 	if (fb->map == MAP_FAILED)
 		goto err_add_fb;
-- 
2.17.1

